File: generators/isaac-generator.js
var isaac = require('isaac');

function IsaacGenerator(seed) {
	this._seed = seed;
	this._isaac = isaac
	this._isaac.seed(seed);
}

IsaacGenerator.prototype.next = function () {
	return Math.round(this._isaac.rand());
}

module.exports = IsaacGenerator;

File: generators/isaac_generator.py
from libs.isaac import Isaac


class IsaacGenerator:
    def __init__(self, **kwargs):
        self._result_bits = kwargs.get('result_bits', False)

        self._isaac = Isaac()
        self._mask = 0
        deg = 1
        for i in xrange(self._result_bits):
            self._mask += deg
            deg *= 2
			
    def next(self):
        value = self._isaac.rand()

        return value & self._mask 
